streaming/java/streaming-runtime/src/main/java/io/ray/streaming/runtime/util/ResourceUtil.java

```diff
@@ -0,0 +1,245 @@
+   * during the recent period being observed
+   */
+  public static double getSystemCpuUtilByMXBean() {
+    return osmxb.getSystemCpuLoad();
+  }
+
+  /**
+   * Get system cpu util by vsar
+   */
+  public static double getSystemCpuUtilByVsar() throws Exception {
+    double cpuUsageFromVsar = 0.0;
+    String[] vsarCpuCommand = {"/bin/sh", "-c", "vsar --check --cpu -s util"};
+    try {
+      Process proc = Runtime.getRuntime().exec(vsarCpuCommand);
+      BufferedInputStream bis = new BufferedInputStream(proc.getInputStream());
+      BufferedReader br = new BufferedReader(new InputStreamReader(bis));
+      String line;
+      List<String> processPidList = new ArrayList<>();
+      while ((line = br.readLine()) != null) {
+        processPidList.add(line);
+      }
+      if (!processPidList.isEmpty()) {
+        String[] split = processPidList.get(0).split("=");
+        cpuUsageFromVsar = Double.parseDouble(split[1]) / 100.0D;
+      } else {
+        throw new IOException("Vsar check cpu usage failed, maybe vsar is not installed.");
+      }
+    } catch (Exception e) {
+      LOG.warn("Failed to get cpu usage by vsar.", e);
+      throw e;
+    }
+    return cpuUsageFromVsar;
+  }
+
+  /**
+   * @returns the system load average for the last minute
+   */
+  public static double getSystemLoadAverage() {
+    return osmxb.getSystemLoadAverage();
+  }
+
+  /**
+   * @return system cpu cores num
+   */
+  public static int getCpuCores() {
+    return osmxb.getAvailableProcessors();
+  }
+
+  /**
+   * Get containers by hostname of address
+   *
+   * @param containers container list
+   * @param containerHosts container hostname or address set
+   * @return matched containers
+   */
+  public static List<Container> getContainersByHostname(
+      List<Container> containers,
+      Collection<String> containerHosts) {
+
+    return containers.stream()
+        .filter(container ->
+            containerHosts.contains(container.getHostname()) ||
+                containerHosts.contains(container.getAddress()))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Get container by hostname
+   *
+   * @param hostName container hostname
+   * @return container
+   */
+  public static Optional<Container> getContainerByHostname(
+      List<Container> containers,
+      String hostName) {
+    return containers.stream()
+        .filter(container -> container.getHostname().equals(hostName) ||
+            container.getAddress().equals(hostName))
+        .findFirst();
+  }
+
+  /**
+   * Get container by id
+   *
+   * @param containerID container id
+   * @return container
+   */
+  public static Optional<Container> getContainerById(
+      List<Container> containers,
+      ContainerId containerID) {
+    return containers.stream()
+        .filter(container -> container.getId().equals(containerID))
+        .findFirst();
+  }
+
+  /**
+   * Get container by node unique id
+   *
+   * @param nodeId node unique id
```

Comment:
Could be two input parameters : containers and nodeId

Reply:
done.